lib/amazonka/CHANGELOG.md

```diff
@@ -3,6 +3,33 @@
 ## [2.0.0](https://github.com/brendanhay/amazonka/tree/2.0.0)
 Released: **?**, Compare: [2.0.0-rc1](https://github.com/brendanhay/amazonka/compare/2.0.0-rc1...2.0.0)
 
+### Major Changes
+
+- The authentication code in `amazonka` got a full rewrite in PR [\#746](https://github.com/brendanhay/amazonka/pull/746).
+
+  - `newEnv` now takes a function of type `EnvNoAuth -> m Env`, which is to fetch credentials in an appropriate manner
+  - The `Credentials` type has been removed, you should instead use the following functions corresponding to the departed constructor. All are exported by `Amazonka.Auth`:
+
+    | Old Name          | New Name                     | Comment                                                                                                                                         |
+    |-------------------|------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
+    | `FromKeys`        | `fromKeys`                   | Explicit access key id/secret access key.                                                                                                       |
+    | `FromSession`     | `fromSession`                | As above, but with session token.                                                                                                               |
+    |                   | `fromTemporarySession`       | As above, but with token with expiry time. Likely not useful.                                                                                   |
+    |                   | `fromKeysEnv`                | Reads `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and optionally `AWS_SESSION_TOKEN`.                                                         |
+    | `FromEnv`         |                              | Removed - used to look up named environment variables for access key id/secret access key/session token/expiry time.                            |
```

Comment:
Please sing out if you need this - it would be easy to add a `fromKeysEnvNamed` function, but then I would feel obliged to do it for every `fromFooEnv` function, which feels a bit odd since nobody really needed _those_.